### PR TITLE
auth json

### PR DESCRIPTION
**Pull request by kevmoo**
- auth: add toJson/fromJson to a number of classes
- googleapis_auth: refactor the libraries to follow conventions

### CODE REVIEW

**Comment by kevmoo** [comment]
This should also improve the score of the package on pub!

File: googleapis_auth/lib/googleapis_auth.dart [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+library googleapis_auth;
+
+export 'src/access_credentials.dart';
+export 'src/access_token.dart';
+export 'src/auth_client.dart';
+export 'src/auth_functions.dart';
+export 'src/client_id.dart';
+export 'src/exceptions.dart';
+export 'src/response_type.dart';
+export 'src/service_account_credentials.dart';
```
Comment (jonasfj) [inline googleapis_auth/lib/googleapis_auth.dart:18]: Could we not expose `clientViaServiceAccount` and `clientViaApiKey` using conditional imports/exports?

I'm not sure of there is anything like ADC for js.. but that could also be awesome.

**Review by jonasfj — APPROVED** [review]
I guess exporting something that works both on js and on VM using conditional imports is hard..

Or at-least finding something useful is hard.. so maybe we just consider that later...

Overall, this is fantastic!

File: googleapis_auth/lib/googleapis_auth.dart (hunk 1 above)
Comment (kevmoo) [reply googleapis_auth/lib/googleapis_auth.dart:18]: I'm confused. Explain?

Comment (jonasfj) [reply googleapis_auth/lib/googleapis_auth.dart:18]: Both `lib/auth_browser.dart` and `lib/auth_io.dart` exports `clientViaApiKey`, so why not export it from `lib/googleapis_auth.dart` with a conditional export..

Comment (kevmoo) [reply googleapis_auth/lib/googleapis_auth.dart:18]: https://github.com/dart-lang/googleapis/issues/198

**Pull request merged**